test/Microsoft.DotNet.Tools.Builder.Tests/IncrementalTests.cs

```diff
@@ -66,38 +64,59 @@ public void TestNoLockFile()
             Assert.Contains("does not have a lock file", buildResult.StdErr);
         }
 
-        [Fact(Skip="https://github.com/dotnet/cli/issues/980")]
         public void TestRebuildChangedLockFile()
         {
 
             var buildResult = BuildProject();
-            AssertProjectCompiled(_mainProject, buildResult);
+            AssertProjectCompiled(MainProject, buildResult);
 
-            var lockFile = Path.Combine(_tempProjectRoot.Path, "project.lock.json");
+            var lockFile = Path.Combine(TempProjectRoot.Path, "project.lock.json");
             TouchFile(lockFile);
 
             buildResult = BuildProject();
-            AssertProjectCompiled(_mainProject, buildResult);
+            AssertProjectCompiled(MainProject, buildResult);
         }
 
-        [Fact(Skip="https://github.com/dotnet/cli/issues/980")]
         public void TestRebuildChangedProjectFile()
         {
 
             var buildResult = BuildProject();
-            AssertProjectCompiled(_mainProject, buildResult);
+            AssertProjectCompiled(MainProject, buildResult);
 
-            TouchFile(GetProjectFile(_mainProject));
+            TouchFile(GetProjectFile(MainProject));
 
             buildResult = BuildProject();
-            AssertProjectCompiled(_mainProject, buildResult);
+            AssertProjectCompiled(MainProject, buildResult);
+        }
+
+        // regression for https://github.com/dotnet/cli/issues/965
+        [Fact]
+        public void TestInputHasSameTimeAsOutput()
+        {
+            var buildResult = BuildProject();
+            AssertProjectCompiled(MainProject, buildResult);
+
+            // set all output items to the same time to make it easier to reason about them
+            var now = DateTime.Now;
+            foreach (var f in Directory.EnumerateFiles(GetCompilationOutputPath()))
+            {
+                File.SetLastWriteTime(f, now);
+            }
+
+            // set an input to have the same last write time as an output item
+            // this should trigger recompilation to account for file systems with second timestamp granularity
+            // (an input file that changed within the same second as the previous outputs should trigger a rebuild)
+            File.SetLastWriteTime(GetProjectFile(MainProject), now);
+
+            buildResult = BuildProject();
+            AssertProjectCompiled(MainProject, buildResult);
```

Comment:
Maybe you should make these asserts extension methods of CommandResult. It would make them to read out much nicer.